### PR TITLE
add getValidActiveBrokerPackageName

### PR DESCRIPTION
**Pull request by p3dr0rv**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java [hunk 1]
```diff
@@ -223,6 +223,47 @@ && verifySignature(authenticator.packageName)) {
...
+    // TODO: Consolidate getCurrentActiveBrokerPackageName and getValidActiveBrokerPackageName into one.
+    public String getValidActiveBrokerPackageName() throws ClientException {
+        final String methodTag = TAG + ":getValidActiveBrokerPackageName";
+
+        String authenticatorPackageName = null;
+        try {
+            final AuthenticatorDescription[] authenticators = AccountManager.get(mContext).getAuthenticatorTypes();
+            Logger.info(methodTag, authenticators.length + " Authenticators registered.");
+            for (AuthenticatorDescription authenticator : authenticators) {
+                if (authenticator.type.equals(BROKER_ACCOUNT_TYPE)) {
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: We can flip this:

(BROKER_ACCOUNT_TYPE.equals(authenticator.type)

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java [hunk 2]
```diff
@@ -223,6 +223,47 @@ && verifySignature(authenticator.packageName)) {
...
+                    authenticatorPackageName = authenticator.packageName;
+                }
+            }
+        } catch (final Exception exception) {
+            final ClientException clientException = new ClientException("", exception.getMessage());
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: Why no error code?

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java [hunk 3]
```diff
@@ -223,6 +223,47 @@ && verifySignature(authenticator.packageName)) {
...
+    public String getValidActiveBrokerPackageName() throws ClientException {
+        final String methodTag = TAG + ":getValidActiveBrokerPackageName";
+
+        String authenticatorPackageName = null;
+        try {
+            final AuthenticatorDescription[] authenticators = AccountManager.get(mContext).getAuthenticatorTypes();
+            Logger.info(methodTag, authenticators.length + " Authenticators registered.");
+            for (AuthenticatorDescription authenticator : authenticators) {
+                if (authenticator.type.equals(BROKER_ACCOUNT_TYPE)) {
+                    authenticatorPackageName = authenticator.packageName;
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: You could also invoke `verifySignatureAndThrow` here and then just return early from the loop

Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: If you do that, then you can probably entirely get rid of the `authenticatorPackageName` variable

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java (hunk 2 above)
Comment (p3dr0rv) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: good catch

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java [hunk 4]
```diff
@@ -223,6 +223,45 @@ && verifySignature(authenticator.packageName)) {
...
+    // TODO: Consolidate getCurrentActiveBrokerPackageName and getValidActiveBrokerPackageName into one.
+    public String getValidActiveBrokerPackageName() throws ClientException {
+        final String methodTag = TAG + ":getValidActiveBrokerPackageName";
+
+        final int numberOfAuthenticators;
+        try {
+            final AuthenticatorDescription[] authenticators = AccountManager.get(mContext).getAuthenticatorTypes();
+            numberOfAuthenticators = authenticators.length;
+            Logger.info(methodTag, numberOfAuthenticators + " Authenticators registered.");
+            for (AuthenticatorDescription authenticator : authenticators) {
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]:
```suggestion
            for (final AuthenticatorDescription authenticator : authenticators) {
```

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java [hunk 5]
```diff
@@ -223,6 +223,45 @@ && verifySignature(authenticator.packageName)) {
...
+                    return authenticator.packageName;
+                }
+            }
+        } catch (final Exception exception) {
+            final ClientException clientException = new ClientException(UNKNOWN_ERROR, exception.getMessage());
+            Logger.error(methodTag, exception.getMessage(), exception);
+            throw clientException;
+        }
+        final String errorMessage = "None of the " + numberOfAuthenticators + " authenticators, is type: " + BROKER_ACCOUNT_TYPE;
+        final ClientException clientException = new ClientException(UNKNOWN_ERROR, errorMessage);
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: We can probably have a more specific error code here I guess that indicates something like BROKER_NOT_INSTALLED

**Review by shahzaibj — APPROVED** [review]

**Pull request merged**